pkg/asset/manifests/topologies.go

```diff
@@ -28,6 +28,11 @@ func determineTopologies(installConfig *types.InstallConfig) (controlPlaneTopolo
 	for _, mp := range installConfig.Compute {
 		numOfWorkers += ptr.Deref(mp.Replicas, 0)
 	}
+	if numOfWorkers < 2 {
+		// Control planes are schedulable when there are < 2 workers.
+		// Adjust the number of schedulable nodes here to reflect.
+		numOfWorkers += controlPlaneReplicas
+	}
```

Comment:
Also, what about `case 1: ` below?

https://github.com/openshift/installer/blob/4180662515e75d11d0a7984c66d574f3c64a765a/pkg/asset/manifests/topologies.go#L42-L43

It is setting infrastructure topology is `configv1.SingleReplicaTopologyMode `. This new check will also make the `case 1: ` unreachable unless SNO.